icx-proxy/src/main.rs

```diff
@@ -418,17 +422,26 @@ async fn handle_request(
     ip_addr: IpAddr,
     request: Request<Body>,
     replica_url: String,
+    proxy_url: Option<String>,
     dns_canister_config: Arc<DnsCanisterConfig>,
     logger: slog::Logger,
     debug: bool,
 ) -> Result<Response<Body>, Infallible> {
-    match if request.uri().path().starts_with("/api/") {
+    let request_uri_path = request.uri().path();
+    match if request_uri_path.starts_with("/api/") {
         slog::debug!(
             logger,
             "URI Request to path '{}' being forwarded to Replica",
             &request.uri().path()
         );
         forward_api(&ip_addr, request, &replica_url).await
+    } else if proxy_url.is_some() && request_uri_path.starts_with("/_/") {
```

Comment:
Need to return 404 if there's no `proxy_url` configured.

Reply:
Ah I see we didn't do that before. I won't block this PR on it then.

Reply:
I'm also not clear on how this is supposed to work outside of `dfx`

Reply:
Anyway, this is updated to return 404 now